lib/plugins/aws/deploy/lib/uploadArtifacts.js

```diff
@@ -18,7 +18,18 @@ module.exports = {
       Body: body,
       ContentType: 'application/json',
     };
-
+    if (this.serverless.service.provider.deploymentBucketOptions) {
```

Comment:
Thanks for looking into this and the proposal @Athulus 

Right now our standard is to have everything `camelCase` in the `serverless.yml` config file. The `resources` section is an exception since the code written there will be directly merged into the CloudFormation template and therefore needs to follow their conventions (which is `UpperCase`)